### PR TITLE
Max & Maya: Make sure validators being shown in the Publisher UI when they set to be optional in AYON setting  

### PR DESCRIPTION
**Pull request by moonyuet**
## Changelog Description
This PR is to make sure validators being shown correctly in the Publisher UI when they are being set to be optional in AYON setting.

## Additional info
n/a

## Testing notes:
1. Launch Max/Maya
2. Create Instances
3. Toggle the validators optional switch in Max/Maya settings
4. Check to see whether the validators are being shown in the Publisher UI

### CODE REVIEW

File: client/ayon_core/hosts/maya/plugins/publish/validate_current_renderlayer_renderable.py [hunk 1]
```diff
@@ -20,9 +20,11 @@ class ValidateCurrentRenderLayerIsRenderable(pyblish.api.ContextPlugin):
     order = pyblish.api.ValidatorOrder
     hosts = ["maya"]
     families = ["renderlayer"]
+    optional = False
 
     def process(self, context):
-
+        if not self.is_active(context.data):
```
Comment (BigRoy) [inline client/ayon_core/hosts/maya/plugins/publish/validate_current_renderlayer_renderable.py:26]: This plugin must still inherit from `OptionalPyblishPluginMixin`

File: client/ayon_core/hosts/maya/plugins/publish/validate_glsl_material.py [hunk 2]
```diff
@@ -23,6 +24,8 @@ class ValidateGLSLMaterial(pyblish.api.InstancePlugin):
     active = True
 
     def process(self, instance):
+        if not self.is_active(instance.data):
```
Comment (BigRoy) [inline client/ayon_core/hosts/maya/plugins/publish/validate_glsl_material.py:27]: This plugin must still inherit from `OptionalPyblishPluginMixin`

File: client/ayon_core/hosts/maya/plugins/publish/validate_loaded_plugin.py [hunk 3]
```diff
@@ -35,7 +36,8 @@ def get_invalid(cls, context):
         return invalid
 
     def process(self, context):
-
+        if not self.is_active(context.data):
```
Comment (BigRoy) [inline client/ayon_core/hosts/maya/plugins/publish/validate_loaded_plugin.py:39]: This plugin must still inherit from `OptionalPyblishPluginMixin`

File: client/ayon_core/hosts/maya/plugins/publish/validate_look_shading_group.py [hunk 4]
```diff
@@ -14,17 +14,20 @@ class ValidateShadingEngine(pyblish.api.InstancePlugin):
...
     hosts = ["maya"]
     label = "Look Shading Engine Naming"
     actions = [
         ayon_core.hosts.maya.api.action.SelectInvalidAction, RepairAction
     ]
+    optional = True
 
     # The default connections to check
     def process(self, instance):
+        if not self.is_active(instance.data):
```
Comment (BigRoy) [inline client/ayon_core/hosts/maya/plugins/publish/validate_look_shading_group.py:29]: This plugin must still inherit from `OptionalPyblishPluginMixin`

**Review by BigRoy — COMMENTED** [review]
I'm personally not a fan of just throwing this into **ALL** validator plug-ins. Some just really aren't meant to ever be optional plus it's more code to maintain - and in the end does more 'logic' per plug-in and thus will be (maybe measurably?) slower to run.

Anyway, I can see how having the consistency might make it 'less error prone' on changes in the future if someone did start exposing one to settings. But half of these plug-ins don't have matching "optional" settings I believe, and thus this logic being implemented now "does nothing meaningful" because the plug-ins are never actually optional.

So I'd recommend just 'fixing' the ones that need fixing - instead of cluttering it across all plug-ins where it doesn't currently do anything useful. **OR** add them all to the customizable settings?

**Comment by m-u-r-p-h-y** [comment]
> So I'd recommend just 'fixing' the ones that need fixing - instead of cluttering it across all plug-ins where it doesn't currently do anything useful. **OR** add them all to the customizable settings?

We are not making any new validator optional here, I suppose. We are just fixing the ones that already have optional tag to propagate it to the publishing window so they are really optional.

image

**Comment by BigRoy** [comment]
> > So I'd recommend just 'fixing' the ones that need fixing - instead of cluttering it across all plug-ins where it doesn't currently do anything useful. **OR** add them all to the customizable settings?
> 
> We are not making any new validator optional here, I suppose. We are just fixing the ones that already have optional tag in settings to propagate it to the publishing window so they are really optional.
> 
> image

Thanks, yes. Kayla just hit me up on discord in a PM as well mentioning these were already in the settings - in that case. Great fixing! 🧑‍🔧

**Review by mkolar — APPROVED** [review]

**Pull request merged**